test/Cimpress.Extensions.Http.Caching.Redis.UnitTests/RedisCacheHandler_when_sending_request.cs

```diff
@@ -158,5 +157,59 @@ public static IEnumerable<object[]> GetHeadData
             testMessageHandler.NumberOfCalls.Should().Be(2);
             cache.Verify(c => c.SetAsync("http://unittest/", It.IsAny<byte[]>(), It.IsAny<DistributedCacheEntryOptions>()), Times.Exactly(0));
         }
+        
+        [Fact]
+        public async Task Invalidates_cache_correctly()
+        {
+            // setup
+            var testMessageHandler = new TestMessageHandler();
+            var cache = new Mock<IDistributedCache>(MockBehavior.Strict);
+            var url = "http://unittest/";
+            var key = HttpMethod.Get + url;
+            var cacheResult = new CacheData(new byte[0], new HttpResponseMessage(HttpStatusCode.OK), null, null).Serialize();
+            var nonCacheResult = default(byte[]);
+            var currentResult = nonCacheResult;
+            cache.Setup(c => c.GetAsync(key)).ReturnsAsync(currentResult);
+            cache.Setup(c => c.RemoveAsync(key)).Returns(Task.FromResult(true)).Callback(() => currentResult = nonCacheResult);
+            cache.Setup(c => c.SetAsync(key, It.IsAny<byte[]>(), It.IsAny<DistributedCacheEntryOptions>())).Returns(Task.FromResult(true))
+                .Callback(() => currentResult = cacheResult);
+
+            var cacheExpirationPerStatusCode = new Dictionary<HttpStatusCode, TimeSpan> {{(HttpStatusCode) 200, TimeSpan.FromHours(1)}};
+
+            var handler = new RedisCacheHandler(testMessageHandler, cacheExpirationPerStatusCode, cache.Object);
+            var client = new HttpClient(handler);
+
+            // execute twice, with cache invalidation in between
+            var uri = new Uri(url);
+            await client.GetAsync(uri);
+            await handler.InvalidateCache(uri, HttpMethod.Get);
+            await client.GetAsync(uri);
+
+            // validate
+            testMessageHandler.NumberOfCalls.Should().Be(2);
+        }
+
+        [Fact]
+        public void Invalidates_cache_per_method()
+        {
+            // setup
+            var testMessageHandler = new TestMessageHandler();
+            var cache = new Mock<IDistributedCache>(MockBehavior.Strict);
+            var url = "http://unittest/";
+            var getKey = HttpMethod.Get + url;
+            var headKey = HttpMethod.Head + url;
+            cache.Setup(c => c.RemoveAsync(headKey)).Returns(Task.FromResult(true));
+            cache.Setup(c => c.RemoveAsync(getKey)).Throws<Exception>();
```

Comment:
Isn't this line unused, if `Get` was invalidated incorrectly we should get a Mock Invocation Exception, right?